packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/object/services/DefaultReadOnlyObjectPredicate.java

```diff
@@ -33,10 +33,19 @@ public boolean test(Object object) {
         boolean result = false;
         if (object instanceof EObject eObject) {
             result = Optional.ofNullable(eObject.eResource())
-                    .map(resource -> resource.eAdapters().stream().anyMatch(LibraryMetadataAdapter.class::isInstance))
+                    .flatMap(resource -> resource.eAdapters().stream()
```

Comment:
Library resources are set as read-only when they are loaded (see `EditingContextDependencyLoader`). 
I don't think we need to check if the resource has the `LibraryMetadataAdapter` here, but we can add it for extra safety.